com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -317,77 +326,69 @@ internal struct AnimationState : INetworkSerializable
             internal float NormalizedTime;
             internal int Layer;
             internal float Weight;
+            internal float Duration;
 
             // For synchronizing transitions
             internal bool Transition;
+            internal bool CrossFade;
+
+            // Flags for bool states
+            private const byte k_IsTransition = 0x01;
+            private const byte k_IsCrossFade = 0x02;
+
+            // Used to serialize the bool states
+            private byte m_StateFlags;
+
             // The StateHash is where the transition starts
             // and the DestinationStateHash is the destination state
             internal int DestinationStateHash;
 
             public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReaderWriter
             {
-                if (serializer.IsWriter)
+                var isWriter = serializer.IsWriter;
+                if (isWriter)
                 {
                     var writer = serializer.GetFastBufferWriter();
-                    var writeSize = FastBufferWriter.GetWriteSize(Transition);
-                    writeSize += FastBufferWriter.GetWriteSize(StateHash);
-                    writeSize += FastBufferWriter.GetWriteSize(NormalizedTime);
-                    writeSize += FastBufferWriter.GetWriteSize(Layer);
-                    writeSize += FastBufferWriter.GetWriteSize(Weight);
+                    m_StateFlags = 0x00;
                     if (Transition)
                     {
-                        writeSize += FastBufferWriter.GetWriteSize(DestinationStateHash);
+                        m_StateFlags |= k_IsTransition;
                     }
-
-                    if (!writer.TryBeginWrite(writeSize))
```

Comment:
This seemed to explicitly want to check it could write the data to error here. Where will it error now for the same condition?

Reply:
The primary reason for the switch was due to the added byte packing to reduce bandwidth costs per update (which the size depends upon how well the property is packed)... the remaining serialized properties didn't make sense, processing wise, to do that same pre-calculation so I just ran with safe serialization for everything so that issue is no longer of concern.